pkg/cluster/kubernetes/talos_managed.go

```diff
@@ -157,12 +159,24 @@ func Upgrade(ctx context.Context, cluster UpgradeProvider, options UpgradeOption
 		}
 	}
 
-	objects, err := getManifests(ctx, cluster)
+	useSSA := false
+
+	semverMinV, err := semver.ParseTolerant(minTalosVersion.String())
+	if err != nil {
+		return fmt.Errorf("failed to parse minimum talos version %q for %w", minTalosVersion.String(), err)
+	}
+
+	talos113version, err := semver.ParseTolerant("v1.13.0-alpha.0")
 	if err != nil {
 		return err
 	}
 
-	return syncManifests(ctx, objects, cluster, options)
+	// Use server-side apply for Talos versions >= v1.13.0
+	if semverMinV.Compare(talos113version) <= 0 {
```

Comment:
looks like the condition is inverted? 

one safe way to compare all pre-release versions without surprises is to get rid of `.Pre` part: https://github.com/siderolabs/talos/blob/b7c49777fe645b85ce9ec72c93922d2b154512f7/pkg/machinery/imager/quirks/quirks.go#L22-L28

and then you can compare with v1.13.0

Reply:
Also, this should be `talos113version.GTE(semverMinV)`

Reply:
refactored to use TalosVersion from compatibility